fuel-asm/src/panic_reason.rs

```diff
@@ -566,3 +165,22 @@ impl From<core::array::TryFromSliceError> for PanicReason {
         Self::MemoryOverflow
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_u8_panic_reason_round_trip() {
+        for i in 0..0x25 {
+            let reason = PanicReason::from(i);
+            let i2 = reason as u8;
+            assert_eq!(i, i2);
+        }
+        for i in 0x25..=255 {
+            let reason = PanicReason::from(i);
+            let i2 = reason as u8;
+            assert_eq!(0x25, i2);
```

Comment:
should this be asserting that the panic reason is `UnknownPanicReason`? Otherwise the test is basically the same besides the ranges.

Reply:
Yes, `0x25` is `UnknownPanicReason` and it checks that any number more than `0x25` will be `UnknownPanicReason`.